src/app/pages/questions/components/question/text-input/text-input.component.ts

```diff
@@ -96,7 +97,14 @@ export class TextInputComponent implements OnInit {
 
   emitAnswer(value) {
     if (typeof value !== 'string') {
-      this.value = Object.assign(this.value, value)
+      if ("date" in value) {
```

Comment:
I think this is doing something similar as `this.value = Object.assign(this.value, value)`?

Reply:
Similar yes, but just doing `this.value = Object.assign(this.value, value)` for the date and time cases etc. will result in simple vectors with theoretically unknown order being emitted. Now the emitted objects are properly named dictionaries. I thought it would be better to have that, since these also get sent as is to the platform, and having ambiguous objects there might lead to problems in analysis later.

Reply:
Ah, I see what you mean. I have changed the wheel values input and output formats so that they use the dictionaries as you mentioned. Let me know what you think. Thanks.

Reply:
Yeah this is much better than what I did!

I had to fix the scroll height, don't know why, it was working fine yesterday and the last relevant change I can see is from 5 days ago.
But this value makes more sense since the height is set to 40 in the scss as well.